src/WebJobs.Script/Config/ActiveHostConfigurationSource.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Extensions.Configuration;
+
+namespace Microsoft.Azure.WebJobs.Script.Configuration
+{
+    public class ActiveHostConfigurationSource : IConfigurationSource
+    {
+        private readonly IScriptHostManager _scriptHostManager;
+
+        public ActiveHostConfigurationSource(IScriptHostManager scriptHostManager)
+        {
+            _scriptHostManager = scriptHostManager;
+        }
+
+        public IConfigurationProvider Build(IConfigurationBuilder builder)
+        {
+            return new ActiveHostConfigurationProvider(_scriptHostManager);
+        }
+
+        private class ActiveHostConfigurationProvider : ConfigurationProvider
+        {
+            private readonly IScriptHostManager _scriptHostManager;
+            private IDisposable _changeTokenRegistration;
+
+            public ActiveHostConfigurationProvider(IScriptHostManager scriptHostManager)
+            {
+                _scriptHostManager = scriptHostManager ?? throw new ArgumentNullException(nameof(scriptHostManager));
+                _scriptHostManager.ActiveHostChanged += ScriptHostManager_ActiveHostChanged;
+            }
+
+            public override void Load()
+            {
+                if ((_scriptHostManager as IServiceProvider)?.GetService(typeof(IConfiguration)) is IConfigurationRoot activeHostConfiguration)
```

Comment:
Avoid the cast here by having your variable be of type `IServiceProvider`, as that is all this code needs.

Reply:
Since we make this ConfigurationProvider implement IDisposable, we'll need the scriptHostManager reference to unregister the event handler and dispose the change token.